> Can you reduce the CI builds to 1 at a time?

Done

@SteveMacenski Sorry for lack of clarification. Here are the build failures that I'm talking about:

```
13:32:14 Scanning dependencies of target _run_tests_camera_calibration_nosetests_test.directed.py
13:32:14 -- run_tests.py: execute commands
13:32:14   "/usr/bin/cmake" -E make_directory /tmp/ws/test_results/camera_calibration
13:32:14   /usr/bin/nosetests3 -P --process-timeout=60 /tmp/ws/src/image_pipeline/camera_calibration/test/directed.py --with-xunit --xunit-file=/tmp/ws/test_results/camera_calibration/nosetests-test.directed.py.xml
13:32:15 FFF.F
13:32:26 ======================================================================
13:32:26 FAIL: test_monocular (directed.TestArtificial)
13:32:26 ----------------------------------------------------------------------
13:32:26 Traceback (most recent call last):
13:32:26   File "/tmp/ws/src/image_pipeline/camera_calibration/test/directed.py", line 258, in test_monocular
13:32:26     self.assert_good_mono(mc, self.limages[i], setup.lin_err)
13:32:26   File "/tmp/ws/src/image_pipeline/camera_calibration/test/directed.py", line 236, in assert_good_mono
13:32:26     self.assertTrue(lin_err < max_err, 'lin_err is %f' % lin_err)
13:32:26 AssertionError: False is not true : lin_err is 0.200393
13:32:26 -------------------- >> begin captured stdout << ---------------------
13:32:26 linear error is 0.200393
13:32:26 
13:32:26 --------------------- >> end captured stdout << ----------------------
13:32:26 
13:32:26 ======================================================================
13:32:26 FAIL: Test that the distortion coefficients returned for a rational_polynomial model are not empty.
13:32:26 ----------------------------------------------------------------------
13:32:26 Traceback (most recent call last):
13:32:26   File "/tmp/ws/src/image_pipeline/camera_calibration/test/directed.py", line 281, in test_rational_polynomial_model
13:32:26     self.assert_good_mono(mc, self.limages[i], setup.lin_err)
13:32:26   File "/tmp/ws/src/image_pipeline/camera_calibration/test/directed.py", line 236, in assert_good_mono
13:32:26     self.assertTrue(lin_err < max_err, 'lin_err is %f' % lin_err)
13:32:26 AssertionError: False is not true : lin_err is 0.215554
13:32:26 -------------------- >> begin captured stdout << ---------------------
13:32:26 linear error is 0.215554
13:32:26 
13:32:26 --------------------- >> end captured stdout << ----------------------
13:32:26 
13:32:26 ======================================================================
13:32:26 FAIL: test_monocular (directed.TestDirected)
13:32:26 ----------------------------------------------------------------------
13:32:26 Traceback (most recent call last):
13:32:26   File "/tmp/ws/src/image_pipeline/camera_calibration/test/directed.py", line 93, in test_monocular
13:32:26     self.assert_good_mono(mc, dim, max_errs[i])
13:32:26   File "/tmp/ws/src/image_pipeline/camera_calibration/test/directed.py", line 82, in assert_good_mono
13:32:26     self.assertTrue(lin_err < max_err, 'lin_err is %f' % lin_err)
13:32:26 AssertionError: False is not true : lin_err is 2.349721
13:32:26 
13:32:26 ======================================================================
13:32:26 FAIL: test_stereo (directed.TestDirected)
13:32:26 ----------------------------------------------------------------------
13:32:26 Traceback (most recent call last):
13:32:26   File "/tmp/ws/src/image_pipeline/camera_calibration/test/directed.py", line 122, in test_stereo
13:32:26     self.assertTrue(epierror < epierrors[i],
13:32:26 AssertionError: False is not true : Epipolar error is 1.745842 for resolution i = 0
13:32:26 -------------------- >> begin captured stdout << ---------------------
13:32:26 Dim = (320, 240)
13:32:26 
13:32:26 Left:
13:32:26 D = [-0.41515829993748576, 0.05110770632388463, 0.07098419147531078, -0.06099314655212798, 0.0]
13:32:26 K = [241.92199869716805, 0.0, 232.55796349899597, 0.0, 249.2727425475938, 35.235295320848486, 0.0, 0.0, 1.0]
13:32:26 R = [0.9792180825079992, 0.07252016617033023, 0.1894010886663131, -0.07248074638854332, 0.9973442246295716, -0.007144158522771803, -0.18941617748311937, -0.006732243063535294, 0.981873815014443]
13:32:26 P = [3033.80782706034, 0.0, 192.16491222381592, 0.0, 0.0, 3033.80782706034, 80.99110150337219, 0.0, 0.0, 0.0, 1.0, 0.0]
13:32:26 
13:32:26 Right:
13:32:26 D = [-0.4195090199363406, 0.05603810767161762, 0.07101175369447646, -0.06441343860299868, 0.0]
13:32:26 K = [241.49812702968313, 0.0, 238.56509880116838, 0.0, 249.06214913696053, 32.67009253686523, 0.0, 0.0, 1.0]
13:32:26 R = [0.9816596131367914, 0.0683023428180517, 0.17798649921185128, -0.06833936111763889, 0.9976445090246253, -0.005930036681791607, -0.17797228901758255, -0.006342206128768554, 0.9840150612481814]
13:32:26 P = [3033.80782706034, 0.0, 192.16491222381592, -285.39522943093715, 0.0, 3033.80782706034, 80.99110150337219, 0.0, 0.0, 0.0, 1.0, 0.0]
13:32:26 self.T = [-0.09234631409917826, -0.006425312317199932, -0.01674347904474744]
13:32:26 self.R = [0.9999229626365073, 0.00423040385007531, 0.011669296272495834, -0.004225813037723542, 0.9999909838333203, -0.00041803856090849416, -0.011670959532113154, 0.0003686940919905222, 0.9999318240601537]
13:32:26 
13:32:26 --------------------- >> end captured stdout << ----------------------
13:32:26 
13:32:26 ----------------------------------------------------------------------
13:32:26 Ran 5 tests in 11.732s
13:32:26 
13:32:26 FAILED (failures=4)
13:32:26 -- run_tests.py: verify result "/tmp/ws/test_results/camera_calibration/nosetests-test.directed.py.xml"
13:32:26 Built target _run_tests_camera_calibration_nosetests_test.directed.py
13:32:26 Scanning dependencies of target _run_tests_camera_calibration_nosetests_test.multiple_boards.py
13:32:26 -- run_tests.py: execute commands
13:32:26   "/usr/bin/cmake" -E make_directory /tmp/ws/test_results/camera_calibration
13:32:26   /usr/bin/nosetests3 -P --process-timeout=60 /tmp/ws/src/image_pipeline/camera_calibration/test/multiple_boards.py --with-xunit --xunit-file=/tmp/ws/test_results/camera_calibration/nosetests-test.multiple_boards.py.xml
13:32:33 F
13:32:33 ======================================================================
13:32:33 FAIL: test_multiple_boards (multiple_boards.TestMultipleBoards)
13:32:33 ----------------------------------------------------------------------
13:32:33 Traceback (most recent call last):
13:32:33   File "/tmp/ws/src/image_pipeline/camera_calibration/test/multiple_boards.py", line 71, in test_multiple_boards
13:32:33     self.assert_(epi_big < 1.0, "Epipolar error for large checkerboard > 1.0. Error: %.2f" % epi_big)
13:32:33 AssertionError: False is not true : Epipolar error for large checkerboard > 1.0. Error: 152.00
13:32:33 -------------------- >> begin captured stdout << ---------------------
13:32:33 
13:32:33 Left:
13:32:33 D = [8.033206324131141, -1112.4968054883295, -0.12464117007090397, 0.05145877700150152, 44346.23866762433]
13:32:33 K = [3859.287985260978, 0.0, 283.5779783443467, 0.0, 3172.76596484317, 346.65918955202807, 0.0, 0.0, 1.0]
13:32:33 R = [0.130008206809717, 0.008478185353019177, 0.9914766696878157, -0.03679333342088533, 0.9993159696919921, -0.0037206631577533767, -0.9908300140679538, -0.03599601494157029, 0.13023121795646708]
13:32:33 P = [10071.06031034245, 0.0, -48972.69860839844, 0.0, 0.0, 10071.06031034245, 1907.7120513916016, 0.0, 0.0, 0.0, 1.0, 0.0]
13:32:33 
13:32:33 Right:
13:32:33 D = [-2.9599126406348053, 371.7125110296346, -0.13320120391531134, 0.09444426126177655, 3.430916753556746]
13:32:33 K = [3948.9941661898433, 0.0, 267.96478042930727, 0.0, 3400.628522793096, 181.75860807601285, 0.0, 0.0, 1.0]
13:32:33 R = [0.12433607052142116, 0.06127832696737718, 0.9903461557512967, -0.03289563728133663, 0.9977970921299045, -0.05760937410666613, -0.9916947204735674, -0.02541514472093939, 0.12607795923025877]
13:32:33 P = [10071.06031034245, 0.0, -48972.69860839844, -3892.741051307987, 0.0, 10071.06031034245, 1907.7120513916016, 0.0, 0.0, 0.0, 1.0, 0.0]
13:32:33 self.T = [-0.0480593016983534, -0.023685754189796043, -0.38279595362357993]
13:32:33 self.R = [0.9999759435587814, 0.0038780665591547437, -0.0057509045799394864, -0.003563507490409819, 0.9985489416209146, 0.05373371941450881, 0.005950942622021858, -0.05371193338089902, 0.9985387396062284]
13:32:33 
13:32:33 --------------------- >> end captured stdout << ----------------------
13:32:33 -------------------- >> begin captured logging << --------------------
13:32:33 rospy.topics: INFO: topicmanager initialized
13:32:33 --------------------- >> end captured logging << ---------------------
13:32:33 
13:32:33 ----------------------------------------------------------------------
13:32:33 Ran 1 test in 6.757s
13:32:33 
13:32:33 FAILED (failures=1)
```

--- a/camera_calibration/src/camera_calibration/calibrator.py
+++ b/camera_calibration/src/camera_calibration/calibrator.py
@@ -381,7 +381,7 @@ class Calibrator():
             num_pts = b.n_cols * b.n_rows
             opts_loc = numpy.zeros((num_pts, 1, 3), numpy.float32)
             for j in range(num_pts):
-                opts_loc[j, 0, 0] = (j / b.n_cols)
+                opts_loc[j, 0, 0] = (j // b.n_cols)
                 if self.pattern == Patterns.ACircles:
                     opts_loc[j, 0, 1] = 2*(j % b.n_cols) + (opts_loc[j, 0, 0] % 2)
                 else:

--- a/image_view/CMakeLists.txt
+++ b/image_view/CMakeLists.txt
@@ -38,19 +38,17 @@ if(ANDROID)
   return()
 endif()
 
-find_package(GTK2)
-add_definitions(-DHAVE_GTK)
-include_directories(${GTK2_INCLUDE_DIRS})
-
 find_package(PkgConfig REQUIRED)
+pkg_check_modules(GTK3 REQUIRED gtk+-3.0)
+include_directories(${GTK3_INCLUDE_DIRS})
+
 pkg_check_modules(PC_HB REQUIRED harfbuzz)
 include_directories(${PC_HB_INCLUDE_DIRS})
 
 # Nodelet library
 add_library(image_view src/nodelets/image_nodelet.cpp src/nodelets/disparity_nodelet.cpp src/nodelets/window_thread.cpp)
 target_link_libraries(image_view ${catkin_LIBRARIES}
-                                 ${GTK_LIBRARIES}
-                                 ${GTK2_LIBRARIES}
+                                 ${GTK3_LIBRARIES}
                                  ${OpenCV_LIBRARIES}
                                  ${Boost_LIBRARIES}
 )
@@ -77,8 +75,7 @@ target_link_libraries(disparity_view ${catkin_LIBRARIES}
 add_executable(stereo_view src/nodes/stereo_view.cpp)
 target_link_libraries(stereo_view ${Boost_LIBRARIES}
                                   ${catkin_LIBRARIES}
-                                  ${GTK_LIBRARIES}
-                                  ${GTK2_LIBRARIES}
+                                  ${GTK3_LIBRARIES}
                                   ${OpenCV_LIBRARIES}
 )
 

--- a/image_view/package.xml
+++ b/image_view/package.xml
@@ -24,7 +24,7 @@
   <build_depend>camera_calibration_parsers</build_depend>
   <build_depend version_gte="1.11.13">cv_bridge</build_depend>
   <build_depend>dynamic_reconfigure</build_depend>
-  <build_depend>gtk2</build_depend>
+  <build_depend>gtk3</build_depend>
   <build_depend>image_transport</build_depend>
   <build_depend>message_filters</build_depend>
   <build_depend>message_generation</build_depend>
@@ -38,7 +38,7 @@
   <run_depend>camera_calibration_parsers</run_depend>
   <run_depend version_gte="1.11.13">cv_bridge</run_depend>
   <run_depend>dynamic_reconfigure</run_depend>
-  <run_depend>gtk2</run_depend>
+  <run_depend>gtk3</run_depend>
   <run_depend>image_transport</run_depend>
   <run_depend>message_filters</run_depend>
   <run_depend>nodelet</run_depend>

